internal/gmm/common/user_profile.go

```diff
@@ -69,6 +69,25 @@ func AttachRanUeToAmfUeAndReleaseOldIfAny(amfUe *context.AmfUe, ranUe *context.R
 	amfUe.AttachRanUe(ranUe)
 }
 
+func AttachRanUeToAmfUeAndReleaseOldHandover(amfUe *context.AmfUe, sourceRanUe, targetRanUe *context.RanUe) {
+	logger.GmmLog.Debugln("In AttachRanUeToAmfUeAndReleaseOldHandover")
+
+	if sourceRanUe != nil {
+		sourceRanUe.DetachAmfUe()
+		if amfUe.T3550 != nil {
+			amfUe.State[targetRanUe.Ran.AnType].Set(context.Registered)
+		}
+		StopAll5GSMMTimers(amfUe)
+		causeGroup := ngapType.CausePresentNas
```

Comment:
Is this a more sutible cause for n2 release?
![image](https://github.com/user-attachments/assets/cd92e215-17bb-4a60-9f7c-d746b25a3fa8)